kernel/comps/keyboard/src/i8042_chip/mouse.rs

```diff
@@ -0,0 +1,472 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! The i8042 mouse driver.
+
+use alloc::{string::String, sync::Arc, vec::Vec};
+use core::{
+    array,
+    sync::atomic::{AtomicBool, Ordering},
+};
+
+use aster_input::{
+    event_type_codes::{KeyCode, KeyStatus, RelCode, SynEvent},
+    input_dev::{InputCapability, InputDevice, InputEvent, InputId, RegisteredInputDevice},
+};
+use ostd::{
+    arch::{
+        irq::{MappedIrqLine, IRQ_CHIP},
+        trap::TrapFrame,
+    },
+    irq::IrqLine,
+    sync::{LocalIrqDisabled, SpinLock},
+};
+use spin::Once;
+
+use super::controller::{
+    I8042Controller, I8042ControllerError, I8042_CONTROLLER, PS2_ACK, PS2_BAT_OK, PS2_CMD_RESET,
+};
+use crate::alloc::string::ToString;
+
+const PACKET_LEN_STANDARD: usize = 3;
+const PACKET_LEN_INTELLIMOUSE: usize = 4;
+const PACKET_LEN_MAX: usize = 4;
+
+/// IRQ line for i8042 mouse.
+static IRQ_LINE: Once<MappedIrqLine> = Once::new();
+
+/// Registered device instance for event submission.
+static REGISTERED_DEVICE: Once<RegisteredInputDevice> = Once::new();
+
+/// ISA interrupt number for i8042 mouse.
+const ISA_INTR_NUM: u8 = 12;
+
+/// Mouse packet state machine.
+static MOUSE_PACKET_STATE: SpinLock<MousePacketState, LocalIrqDisabled> =
+    SpinLock::new(MousePacketState::new());
+
+trait MouseCommand {
+    const CMD_BYTE: u8;
+    const DATA_LEN: usize;
+    const RES_LEN: usize;
+}
+
+mod mouse_cmd {
+    use super::MouseCommand;
+
+    macro_rules! define_commands {
+        (
+            $(
+                $name:ident, $cmd:literal, fn([u8; $dlen:literal]) -> [u8; $rlen:literal];
+            )*
+        ) => {
+            $(
+                pub(super) struct $name;
+                impl MouseCommand for $name {
+                    const CMD_BYTE: u8 = $cmd;
+                    const DATA_LEN: usize = $dlen;
+                    const RES_LEN: usize = $rlen;
+                }
+            )*
+        };
+    }
+
+    define_commands! {
+        GetDeviceId, 0xF2, fn([u8; 0]) -> [u8; 1];
+        SetSampleRate, 0xF3, fn([u8; 1]) -> [u8; 0];
+        EnableDataReporting, 0xF4, fn([u8; 0]) -> [u8; 0];
+    }
+}
+use mouse_cmd::*;
```

Comment:
Using `cmd::GetDeviceId` is more clearer than a global import? The `mouse_` prefix is not really necessary because we're in the `mouse.rs` module.

In fact, we can remove it from some type names as well, e.g., `MouseInitCtx` -> `InitCtx`, `MousePacketState` -> `PacketState`.